llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -800,6 +800,13 @@ instCombineConvertFromSVBool(InstCombiner &IC, IntrinsicInst &II) {
 
 static std::optional<Instruction *> instCombineSVESel(InstCombiner &IC,
                                                       IntrinsicInst &II) {
+  // svsel(ptrue, x, y) => x
+  auto *OpPredicate = II.getOperand(0);
```

Comment:
This looks like a valid optimisation, but it also suggests that the ACLE code written in C/C++ would benefit from being rewritten in a way that avoids this.

Reply:
Yes, you are right. Writing more efficient code is another way to optimize.